apps/api/tests/Animals.spec.ts

```diff
@@ -0,0 +1,201 @@
+      data: NewPetData
+    });
+    expect(postPetRequest).toBeOK();
+    const postPetRequestData = await postPetRequest.json();
+
+    // GET Request to compare with
+    const getNewPetResponse = await apiContext.get(`${url}/animals/${postPetRequestData.id}`);
+    const getNewPetResponseData = await getNewPetResponse.json();
+    const expectedData = {
+      ...NewPetData,
+      id: postPetRequestData.id,
+      birthdate: postPetRequestData.birthdate
+    }
+    expect(getNewPetResponseData).toEqual(expectedData);
+  });
+
+  //POST Add New Pet Fails
+  test("POST usign the incorrect data should return 400", async () => {
+    const postPetRequest = await apiContext.post(`${url}/animals`, {
+      data: NewPetDataFail
+    });
+    expect(postPetRequest.status()).toBe(400);
+  });
+
+  test("POST usign invalid parameters should return 400", async () => {
+    const postPetRequest = await apiContext.post(`${url}/animals`, {
+      data: NewPetDataSetFailWrongParameter
+    });
+    expect(postPetRequest.status()).toBe(400);
+  });
+
+  test("POST usign the incorrect resource should return 404", async () => {
+    const postPetRequest = await apiContext.post(`${url}/animalls`, {
+      data: NewPetData
+    });
+    expect(postPetRequest.status()).toBe(404);
+  });
+
+  test("POST usign an empty data set should return 400", async () => {
+    const postPetRequest = await apiContext.post(`${url}/animals`, {
+      data: NewPetDataFailEmpty
+    });
+    expect(postPetRequest.status()).toBe(400);
+  });
+
+  //PUT (Update) Pets Pass
+  test("PUT using the correct data should update a pet", async () => {
+    const PutUpdatedPet = await apiContext.put(`${url}/animals/2`,
+      {
+        data: NewPetDataUpdate
+      });
+    expect(PutUpdatedPet.ok()).toBeTruthy();
+    const putPetRequestData = await PutUpdatedPet.json();
+
+    // GET Request to compare with
+    const getNewPetResponse = await apiContext.get(`${url}/animals/${putPetRequestData.id}`);
+    const getNewPetResponseData = await getNewPetResponse.json();
+    const expectedData = {
+      ...NewPetDataUpdate,
+      id: putPetRequestData.id,
+      birthdate: putPetRequestData.birthdate
+    }
+    expect(getNewPetResponseData).toEqual(expectedData);
+  });
+
+  //PUT (Update) Pets Fails
+  test("PUT using the incorrect resource should return 404", async () => {
+    const PutUpdatedPet = await apiContext.put(`${url}/animalls/2`,
+      {
+        data: NewPetDataUpdate
+      });
+      expect(PutUpdatedPet.status()).toBe(404);
+  });
+  
+  test("PUT using the incorrect data should return 400", async () => {
+    const PutUpdatedPet = await apiContext.put(`${url}/animals/2`,
+      {
+        data: NewPetDataFail
+      });
+      expect(PutUpdatedPet.status()).toBe(400);
+  });
+
+  test("PUT using invalid parameters should return 400", async () => {
+    const PutUpdatedPet = await apiContext.put(`${url}/animals/2`,
+      {
+        data: NewPetDataSetFailWrongParameter
```

Comment:
```suggestion
        data: NewPetWithInvalidParameter
```
Same as above, just a slight name change of the file and variable name to keep it more consistent as the previous file but also descriptive of the content.

Reply:
Done, thanks!